app/controllers/account_transactions_controller.rb

```diff
@@ -0,0 +1,59 @@
+class AccountTransactionsController < ApplicationController
+  before_action :authenticate_user
+
+  def new
+    @payment_form = PaymentForm.new(
+      account_transaction: transaction_to_pay,
+      coupon: coupon,
+      user: current_user
+    )
+  end
+
+  def create
+    @payment_form = PaymentForm.new(
+      account_transaction: transaction_to_pay,
+      coupon: coupon,
+      user: current_user
+    )
+
+    # do some authorization checks
+
+    if @payment_form.save
+      redirect_to root_path, notice: 'Payment successfully received'
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def account_transaction_params
+    params.require(:account_transaction).permit(:processor_token, :coupon_code)
+  end
+
+  def authenticate_user
+    return if user_signed_in?
+
+    sign_in(
+      User.first_or_create(first_name: 'John', last_name: 'Lennon', email: 'john@example.com')
+    )
```

Comment:
☠️ why don't you let authentication out of this example? Even if it doesn't make sense to pay without a user?